packages/@tailwindcss-upgrade/src/template/codemods/arbitrary-value-to-bare-value.ts

```diff
@@ -0,0 +1,94 @@
+import type { Config } from 'tailwindcss'
+import { parseCandidate, type Candidate, type Variant } from '../../../../tailwindcss/src/candidate'
+import type { DesignSystem } from '../../../../tailwindcss/src/design-system'
+import { segment } from '../../../../tailwindcss/src/utils/segment'
+import { printCandidate } from '../candidates'
+
+export function arbitraryValueToBareValue(
+  designSystem: DesignSystem,
+  _userConfig: Config,
+  rawCandidate: string,
+): string {
+  for (let candidate of parseCandidate(rawCandidate, designSystem)) {
+    let clone = structuredClone(candidate)
+    let changed = false
+    for (let variant of variants(clone)) {
+      // Convert `data-[selected]` to `data-selected`
+      if (
+        variant.kind === 'functional' &&
+        variant.root === 'data' &&
+        variant.value?.kind === 'arbitrary' &&
+        !variant.value.value.includes('=')
+      ) {
+        changed = true
+        variant.value = {
+          kind: 'named',
+          value: variant.value.value,
+        }
+      }
+
+      // Convert `aria-[selected="true"]` to `aria-selected`
+      else if (
+        variant.kind === 'functional' &&
+        variant.root === 'aria' &&
+        variant.value?.kind === 'arbitrary' &&
+        (variant.value.value.endsWith('=true') ||
+          variant.value.value.endsWith('="true"') ||
+          variant.value.value.endsWith("='true'"))
+      ) {
+        let [key, _value] = segment(variant.value.value, '=')
+        if (
+          // aria-[foo~="true"]
+          key[key.length - 1] === '~' ||
+          // aria-[foo|="true"]
+          key[key.length - 1] === '|' ||
+          // aria-[foo^="true"]
+          key[key.length - 1] === '^' ||
+          // aria-[foo$="true"]
+          key[key.length - 1] === '$' ||
+          // aria-[foo*="true"]
+          key[key.length - 1] === '*'
```

Comment:
I had a regex that covered all of these cases but it was horrible so made it more explicit 🙈

Reply:
lol its almost all special regex chars 😂  — I don't think any of them need to be escaped in a character class tho right? (unless `^` was listed first iirc). I think its this right `/[~|^$*]$/.test(key)`?

Reply:
explicit list of cases is fine though. It definitely reads better.

Reply:
This is the one I used originally `/[^~|^$*]=(['"]?)true\1/`

Reply:
ahhh yeah that is kinda awful lol